### PR TITLE
Endpoint for loading campaign donation history

### PR DESCRIPTION
**Pull request by aironhight**
[[Campaign] Create endpoint for loading public campaign donation history #67](https://github.com/podkrepi-bg/api/issues/67)

### CODE REVIEW

**Review by kachar — APPROVED** [review]
This endpoint looks good @aironhight 

I was a bit curious about the use case with no available person, feels okay, but we modify db objects with hardcoded strings, maybe we need another approach to solve this.

On the frontend it's not going to be clear how to distinguish the anonymous from the real person beside checking the first name

**Comment by aironhight** [comment]
I do the check for no available person, because in the DB schema it is shown as nullable.
My thoughts were - if there is no person object associated with the donation, the donation is anonymous.

image

**Comment by kachar** [comment]
@aironhight Yep, that exactly right, but then why do we create a fake person and attach it to the response?

File: apps/api/src/campaign/campaign.service.ts [hunk 1]
```diff
@@ -116,6 +116,40 @@ export class CampaignService {
...
+      },
+    })
+
+    donations.map((donation) => {
+      if (!donation.person)
+        donation.person = { firstName: 'anonymous', lastName: '' }
+    })
+
+    return donations
+  }
```
Comment (igoychev) [inline apps/api/src/campaign/campaign.service.ts:151]: the donations can be extracted from the database using a single query in this way:

```
const donations = await this.prisma.donation.findMany({
      where: { 
        status: 'succeeded',  //this is could be passed in the request too
        targetVault: {
          campaignId: { equals: campaignId}
        }
      },
      include: {
        person: { select: { firstName: true, lastName: true } },
      },
    })
```

and it generates empty array if there are no donations or no campaignId.

Example response:
`[
  {
    "id": "15cb9df5-ba2a-4d91-b409-3392a3d7348d",
    "type": "donation",
    "status": "succeeded",
    "provider": "none",
    "targetVaultId": "0ca7229e-af8b-4409-8c05-48e3318591ec",
    "extCustomerId": "",
    "extPaymentIntentId": "",
    "extPaymentMethodId": "",
    "createdAt": "2022-04-04T07:22:03.608Z",
    "updatedAt": "2022-04-04T07:30:14.787Z",
    "amount": 111,
    "currency": "BGN",
    "personId": null,
    "person": null
  }
]`

**Pull request merged**